pkgs/development/python-modules/aria2p/default.nix

```
@@ -0,0 +1,43 @@
+{ stdenv, buildPythonPackage, fetchFromGitHub, pythonOlder
+, aria2, poetry, pytest, pytestcov, pytest_xdist, responses
+, asciimatics, loguru, requests, setuptools, websocket_client
+}:
+
+buildPythonPackage rec {
+  pname = "aria2p";
+  version = "0.7.0";
```

there's a special builder for poetry
```suggestion
  version = "0.7.0";
  format = "pyproject";
```

---

The `poetry as a top-level package` happened recently, but agreed, it should be updated. The only reason why i know it exists is because I read through the source.